feat: storage backup (#417)
web3-storage/web3.storage: packages/api/README.md

    wrangler secret put FAUNA_KEY --env $(whoami) # Get from fauna.com after creating a dev Classic DB
    wrangler secret put CLUSTER_BASIC_AUTH_TOKEN --env $(whoami) # Get from web3.storage vault in 1password (not required for dev)
    wrangler secret put SENTRY_DSN --env $(whoami) # Get from Sentry (not required for dev)

>     wrangler secret put S3_BUCKET_REGION --env $(whoami) # Get from Amazon S3 (not required for dev)

Reviewer: what's the transition plan from setting a bucket region to picking dynamically? 

Should this be a list of available bucket regions available for production?
Should we just tell devs to create one in a specific region, and in dev mode always default to that region?

Author: Good question. For having multiple regions (from web3.storage put data perspective) we would need multiple S3_BUCKET_REGION and multiple S3_BUCKET_NAME.

What if we move this into wrangler config env vars instead (keeping just the access key as secrets), making them an array? We would only support one for now, but it would be easier to just add more later on.

Either way, with our current approach of parallel cluster.add, we will not have gains from multiple region put. So, the best approach at this point for a next step would be to set up [Amazon S3 Cross-Region Replication (CRR)](https://aws.amazon.com/s3/features/replication). This means we would put it in one region (sometimes not the ideal region for users, but faster than cluster to resolve) and than content would be replicated in other regions to allow enhanced GETs.

Reviewer: > What if we move this into wrangler config env vars instead (keeping just the access key as secrets), making them an array

agree, something like:
```
S3_BUCKETS = [{ s3Region: "xxx", bucketName: }]
```

> this means we would put it in one region.

I think the future enhancement is more like we'd want to put to the single bucket nearest to the cf worker that is receiving the upload, rather than put to many at once.

It'd be neat to have it so that we can have the option in the future of using a set of buckets, 1 per [dvd region](https://meta.wikimedia.org/wiki/List_of_countries_by_regional_classification#/media/File:DVD-Regions_with_key-2.svg) or [whatever](https://meta.wikimedia.org/wiki/List_of_countries_by_regional_classification), and then map cf regions to aws regions, so the put doesn't have to go so far.

I think cross-region replication might be interesting for a cdn, but less so for backing up our backup.

All this said tho, let's leave this PR as is, and then see what we actually need when we get there.

Author: Yeah, that sounds good. It is an easy change later on

>     wrangler secret put S3_BUCKET_REGION --env $(whoami) # Get from Amazon S3 (not required for dev)

Reviewer: ```suggestion
    wrangler secret put S3_BUCKET_REGION --env $(whoami) # e.g us-east-2 (not required for dev)
```

    wrangler secret put S3_ACCESS_KEY_ID --env $(whoami) # Get from Amazon S3 (not required for dev)
    wrangler secret put S3_SECRET_ACCESS_KEY_ID --env $(whoami) # Get from Amazon S3 (not required for dev)

>     wrangler secret put S3_BUCKET_NAME --env $(whoami) # Get from Amazon S3 (not required for dev)

Reviewer: ```suggestion
    wrangler secret put S3_BUCKET_NAME --env $(whoami) # e.g web3.storage-staging-us-east-2 (not required for dev)
```

```

Production vars should be set in Github Actions secrets.

## S3 Setup

> We use [S3](https://aws.amazon.com/s3/) for backup storage. For production an accounton AWS needs to be created.

Reviewer: ```suggestion
We use [S3](https://aws.amazon.com/s3/) for backup storage. For production an account on AWS needs to be created.
```
